boskos/mason/mason.go

```diff
@@ -111,15 +112,19 @@ func NewMason(cleanerCount int, client boskosClient, waitPeriod, syncPeriod time
 }
 
 // CheckUserData helps with extracting leased resource data from the resource.
-func CheckUserData(res common.Resource) (common.LeasedResources, error) {
+func CheckUserData(res common.Resource) (interface{}, error) {
```

Comment:
Rather than making `CheckUserData` return an interface (and causing lots of code paths to have to handle the two types of return values), why not just make this function always return the map (i.e. a `LeasedResources`).

For legacy dynamic objects (without the map), rather than returning the slice, you could instead put all of the resources under a dummy type (e.g. empty string or some invalid-type constant like `-LegacyResources-`). It looks like only the `Destruct` methods actually need the types, right? So we could just limit the error handling to them.

Thoughts?

Reply:
(it might also be worth having this return both the map-by-type as well as the list of all resources, since the list still seems to get used quite a lot.)

Reply:
Yeah, that's much cleaner. Done.